Bibliography
************

.. Note::

   If you use auto-phylo, please cite this publication:
   
   H. López-Fernández; M. Pinto; C.P. Vieira; P. Duque; M. Reboiro-Jato; J. Vieira (2023) `Auto-phylo: A Pipeline Maker 
   for Phylogenetic Studies <https://doi.org/10.1007/978-3-031-38079-2_3>`_. 17th International Conference on Practical Applications of Computational Biology & Bioinformatics: 
   PACBB 2023. Guimarães, Portugal. 12 - July

1.	López-Fernández, H., et al., SEDA: a desktop tool suite for FASTA files processing. IEEE/ACM Transactions on Computational Biology and Bioinformatics, 2020. 19(3): p. 1850-1860.
2.	Sievers, F. and D.G. Higgins, Clustal omega. Current protocols in bioinformatics, 2014. 48(1): p. 3.13. 1-3.13. 16.
3.	Rice, P., I. Longden, and A. Bleasby, EMBOSS: the European molecular biology open software suite. Trends in genetics, 2000. 16(6): p. 276-277.
4.	Abascal, F., R. Zardoya, and M.J. Telford, TranslatorX: multiple alignment of nucleotide sequences guided by amino acid translations. Nucleic acids research, 2010. 38(suppl_2): p. W7-W13.
5.	Price, M.N., P.S. Dehal, and A.P. Arkin, FastTree: computing large minimum evolution trees with profiles instead of a distance matrix. Molecular biology and evolution, 2009. 26(7): p. 1641-1650.
6.	Bettisworth, B. and A. Stamatakis, Root Digger: a root placement program for phylogenetic trees. BMC bioinformatics, 2021. 22(1): p. 1-20.
7.	Darriba, D., et al., jModelTest 2: more models, new heuristics and parallel computing. Nature methods, 2012. 9(8): p. 772-772.
8.	Huelsenbeck, J.P. and F. Ronquist, MRBAYES: Bayesian inference of phylogenetic trees. Bioinformatics, 2001. 17(8): p. 754-755.
9.	Kumar, S., et al., MEGA-CC: computing core of molecular evolutionary genetics analysis program for automated and iterative data analysis. Bioinformatics, 2012. 28(20): p. 2685-2686.
10.	Notredame, C., D.G. Higgins, and J. Heringa, T-Coffee: A novel method for fast and accurate multiple sequence alignment. Journal of molecular biology, 2000. 302(1): p. 205-217.